File: backend/routes/ask.py
from flask import Blueprint, request, jsonify
import requests


API_KEY = 'sec_IogpAHJtfhdFFAnZoXd4y1W3BDGmlt84'


ask_question = Blueprint('ask_question', __name__)


@ask_question.route('/ask', methods=['POST'])
def ask_question_func():
    data = request.json
    source_id = data['sourceId']
    question = data['question']

    url = 'https://api.chatpdf.com/v1/chats/message'
    headers = {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json',
    }
    data = {
        'sourceId': source_id,
        'messages': [
            {
                'role': 'user',
                'content': question,
            }
        ]
    }
    response = requests.post(url, headers=headers, json=data)

    if response.status_code == 200:
        return jsonify({'content': response.json()['content']})
    else:
        return jsonify({'error': response.text}), response.status_code

File: backend/express/routes/ask.js
const express = require('express');
const router = express.Router();
const axios = require('axios');
const { CHATPDF_API_KEY } = require('../config');
const { isAuthenticated } = require('../middleware/authMiddleware');
const Document = require('../models/Document');

router.post('/', isAuthenticated, async (req, res) => {
  try {
    const { sourceId, question } = req.body;

    if (!sourceId || !question) {
      return res.status(400).json({ 
        error: 'sourceId and question are required' 
      });
    }

    // Verificar que el documento pertenece al usuario
    const document = await Document.findOne({
      sourceId,
      userId: req.session.userId
    });

    if (!document) {
      return res.status(404).json({ 
        error: 'Document not found or unauthorized' 
      });
    }

    // Hacer la pregunta a ChatPDF
    const response = await axios.post(
      'https://api.chatpdf.com/v1/chats/message',
      {
        sourceId,
        messages: [{ role: 'user', content: question }]
      },
      {
        headers: {
          'x-api-key': CHATPDF_API_KEY,
          'Content-Type': 'application/json'
        }
      }
    );

    res.json({ 
      content: response.data.content,
      sourceId: sourceId
    });

  } catch (error) {
    console.error('Ask error:', error);
    res.status(500).json({ 
      error: error.message,
      details: error.response?.data || 'No additional details'
    });
  }
});

module.exports = router;
